packages/server/lib/controllers/connection.controller.ts

```diff
@@ -397,6 +405,69 @@ class ConnectionController {
                     connectionCreatedHook: connCreatedHook
                 });
 
+                if (imported) {
+                    updatedConnection = imported;
+                }
+            } else if (template.auth_mode === 'OAUTH2_CC') {
+                const { access_token, oauth_client_id_override, oauth_client_secret_override, expires_at } = req.body;
+
+                if (!access_token) {
+                    errorManager.errRes(res, 'missing_access_token');
+                    return;
+                }
+
+                const { expires_at: parsedExpiresAt } = connectionService.parseRawCredentials(
+                    { access_token, expires_at },
+                    template.auth_mode
+                ) as OAuth2ClientCredentials;
```

Comment:
I used the same pattern used with other oauth mode but using `parseRawCredentials` just to keep the expires_at seems a bit dirty to me